include/cudaq/Optimizer/Transforms/Passes.td

```diff
@@ -576,16 +576,16 @@ def LowerToCFG : Pass<"lower-to-cfg", "mlir::func::FuncOp"> {
   let constructor = "cudaq::opt::createLowerToCFGPass()";
 }
 
-def MappingPass: Pass<"qubit-mapping", "mlir::func::FuncOp"> {
+def MappingFunc: Pass<"qubit-mapping-func", "mlir::func::FuncOp"> {
   let summary = "Perform qubit mapping to account for connectivity constraints";
   let description = [{
     Some backends cannot support any-to-any multi-qubit operations, so this pass
     performs mapping of the qubits by inserting the necessary swap operations
     into the Quake IR.
 
-    Note 1: this pass requires strictly value semantics for any quantum
-    operations. It will throw an error if any memory reference semantics are
-    present on quantum operations.
+    Note 1: this pass requires strictly value semantics (in the form of wire
+    sets) for any quantum operations. It will throw an error if any memory
```

Comment:
How much complexity and effort is it for mapping to be flexible here? That is, it might take null_wire, borrow_wire, or a combination of the two?

Reply:
> How much complexity and effort is it for mapping to be flexible here? That is, it might take null_wire, borrow_wire, or a combination of the two?

I think that would be opening a can of worms that we might not want to open. It would have infer IDs for some of the qubits but not all of them. Some questions you might have to answer are: where do the inferred IDs start? It might be easy for cases like this:

```mlir
%0 = quake.borrow_wire @wires[0] : !quake.wire
// This would probably infer ID 1
%1 = quake.null_wire
```
But it would quickly get ill-defined. For example, consider this case
```mlir
%0 = quake.borrow_wire @wires[1] : !quake.wire
// Would this infer ID 0 or ID 2?
%1 = quake.null_wire
```
And even worse for cases like this:
```mlir
%0 = quake.borrow_wire @wires[2] : !quake.wire
// Would this infer ID 1 or ID 3? Or something else?
%1 = quake.null_wire
%0 = quake.borrow_wire @wires[0] : !quake.wire
```
So I think we should take the opportunity to take the simple path here.

Reply:
So, if we take the combination off the table, then this will work either strictly borrow_wire or strictly null_wire?

Reply:
> So, if we take the combination off the table, then this will work either strictly borrow_wire or strictly null_wire?

No, not as written. I thought wire sets were the "new way" and were strictly better because they were more precise, so I was not intending for anybody to pass null wires into this pass anymore.

Reply:
Just to capture the discussion from the meeting: the mapping pass will require that the input IR be in value semantics form using wire set notation on entry.  The wire set will be defined to correspond to the target machine's model. This should mean a topology was already selected, the wire set has an accurate description of the total number of qubits, $n$, and the qubit borrowing shall never be greater than $n$. If more than $n$ qubits are required, the compilation will terminate with an error.

For the mapper pass proper, if there are any `quake.alloca` or `quake.null_wire`, the above implies we can just emit an error and stop processing.